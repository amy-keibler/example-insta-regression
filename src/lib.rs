use serde::Serialize;

#[derive(Debug, Serialize, PartialEq)]
#[serde(rename_all = "camelCase")]
enum Example {
    ChoiceA(ChoiceAStruct),
    ChoiceB(String),
}

#[derive(Debug, Serialize, PartialEq)]
#[serde(rename_all = "camelCase")]
struct ChoiceAStruct {
    field_a: String,
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn serializes_to_json_correctly() {
        insta::assert_json_snapshot!(vec![
            Example::ChoiceA(ChoiceAStruct {
                field_a: "a".to_string()
            }),
            Example::ChoiceB("b".to_string()),
        ]);
    }
}
