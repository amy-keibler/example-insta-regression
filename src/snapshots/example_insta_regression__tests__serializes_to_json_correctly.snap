---
source: src/lib.rs
expression: "vec![Example :: ChoiceA(ChoiceAStruct { field_a : \"a\".to_string() }), Example\n    :: ChoiceB(\"b\".to_string()),]"
---
[
  {
    "choiceA": {
      "fieldA": "a"
    }
  },
  {
    "choiceB": "b"
  }
]
